src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -167,11 +186,15 @@ public static void UpdateSubscriptionIdForAllResource(this CodeModel codeModel)
                         {
                             setSubParam = true;
                             p.Implementation = ImplementationLocation.Method;
+                            // TODO -- change the operation name to `op.OperationId` when it is ready
+                            MgmtContext.Report.Add(ReportLevel.Information, $"The implementation location of parameter {p.Language.Default.Name} in operation `{operationGroup.Key}_{op.Language.Default.Name}` has been changed to Method");
                         }
                         // update the apiVersion parameter to be 'client' method
                         if (p.Language.Default.Name.Equals("apiVersion", StringComparison.OrdinalIgnoreCase))
                         {
                             p.Implementation = ImplementationLocation.Client;
+                            // TODO -- change the operation name to `op.OperationId` when it is ready
+                            MgmtContext.Report.Add(ReportLevel.Information, $"The implementation location of parameter {p.Language.Default.Name} in operation `{operationGroup.Key}_{op.Language.Default.Name}` has been changed to Client");
```

Comment:
The original location for apiVersion is Client for most cases, only in a few cases like [ResourcesRestOperations](https://github.com/Azure/azure-sdk-for-net/blob/1e11c266329079133cdb62c7fe9738068002c529/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Custom/RestOperations/ResourcesRestOperations.cs#L253), the location was Method (The customized code actually moves back the location for apiVersion to Method to offset the change here. There should be a better example for why the change is made here, but I cannot remember now).

Reply:
GenericResource CRUD need the api-version to be a method parameter, I believe this should be the only use case for this.

I will add a if-check here only to output the log line when there is a change.
Or I could introduce a method in the `Report` class, to determine whether we need the output or not.